### PR TITLE
[TEST] Add unit test cases and TCM config file

### PR DESCRIPTION
**Pull request by gichan-jang**
- Add unit tests for edge sensor using GTest.
 - Add TCM config file.

Signed-off-by: gichan <gichan2.jang@samsung.com>

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -4,3 +4,18 @@ TARGET_INCLUDE_DIRECTORIES(test_edge_sensor PRIVATE ${EDGE_REQUIRE_PKGS_INCLUDE_
 TARGET_LINK_LIBRARIES(test_edge_sensor -L${PROJECT_BINARY_DIR}/src edge-sensor)
 
 INSTALL (TARGETS test_edge_sensor DESTINATION ${BIN_INSTALL_DIR})
+
+# GoogleTest requires at least C++11
+SET(CMAKE_CXX_STANDARD 11)
```
Comment (jaeyun-jung) [inline tests/CMakeLists.txt:9]: let's consider to set 14 as current nnstreamer requires it.
https://github.com/nnstreamer/nnstreamer/blob/50d82aa029c72053004cc35743ba54a79560bc0d/meson.build#L12

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -4,3 +4,18 @@ TARGET_INCLUDE_DIRECTORIES(test_edge_sensor PRIVATE ${EDGE_REQUIRE_PKGS_INCLUDE_
...
+
+INCLUDE(FetchContent)
+FetchContent_Declare(
+  googletest
+  URL https://github.com/google/googletest/archive/609281088cfefc76f9d0ce82e1ff6c30cc3591e5.zip
+)
+
+FetchContent_MakeAvailable(googletest)
```
Comment (again4you) [inline tests/CMakeLists.txt:17]: Since gbs does not support the network, so it would be blocked for a while and failed.
Please remove this line.

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -4,3 +4,18 @@ TARGET_INCLUDE_DIRECTORIES(test_edge_sensor PRIVATE ${EDGE_REQUIRE_PKGS_INCLUDE_
...
+
+ADD_EXECUTABLE(unittest_edge_sensor unittest_edge_sensor.cc)
+TARGET_INCLUDE_DIRECTORIES(unittest_edge_sensor PRIVATE ${INCLUDE_DIR})
+TARGET_LINK_LIBRARIES(unittest_edge_sensor gtest pthread edge-sensor)
```
Comment (again4you) [inline tests/CMakeLists.txt:21]: Instead of using the linked library names directly, what about this?
```cmake
TARGET_LINK_LIBRARIES(unittest_edge_sensor ${TEST_REQUIRE_PKGS_LDFLAGS} edge-sensor)
```
To use this way, you need to update the `CMakefiles.txt` as below.
```cmake
  SET(TEST_REQUIRES_LIST
      gtest
  )
  INCLUDE(FindPkgConfig)
  PKG_CHECK_MODULES(TEST_REQUIRE_PKGS REQUIRED ${TEST_REQUIRES_LIST})
```

In case of Tizen, `BuildRequires:  gtest-devel` in spec is required.

**Review by again4you — CHANGES_REQUESTED** [review]
Please submit the next version. :)

File: tests/CMakeLists.txt (hunk 1 above)
Comment (gichan-jang) [reply tests/CMakeLists.txt]: Good! Updated.

File: tests/CMakeLists.txt (hunk 3 above)
Comment (gichan-jang) [reply tests/CMakeLists.txt]: Thanks! Updated.

File: packaging/nnstreamer-edge.spec [hunk 4]
```diff
@@ -12,6 +12,7 @@ Source1001: nnstreamer-edge.manifest
 
 BuildRequires:  cmake
 BuildRequires:  pkgconfig(paho-mqtt-c)
+BuildRequires:  gtest-devel
```
Comment (again4you) [inline packaging/nnstreamer-edge.spec:15]: What about this?

```spec
%if 0%{?sensor_test}
BuildRequires:  gtest-devel
%endif
```

Comment (gichan-jang) [reply packaging/nnstreamer-edge.spec:16]: It's better! Updated.

**Comment by again4you** [comment]
LGTM!

**Review by again4you — APPROVED** [review]
